docs/federation.md

```diff
@@ -0,0 +1,126 @@
+  - status: federated to all valid clusters successfully
+    type: GDP Federation
+```
+1. `namespace`: namespace of this object must be `avi-system`.
+2. `isLeader`: Users must specify whether the AMKO in the current cluster is leader. Default value is `false`. If set to `false`, AMKO won't sync any objects to the Avi Controller, and the AMKO federator won't federate the objects to the member clusters.
+3. `clusterContext`: Users must specify the current cluster's context. Providing the wrong cluster context can cause undefined behavior.
+4. `version`: Current cluster's AMKO version. If installed via helm, this field gets automatically filled up.
+5. `clusters`: Member cluster list on which federation will be performed. Current cluster (if present) in this list will be ignored.
+6. `status`: Indicates the current state of federation. Following types are reflected in the status:
+  * `current AMKOCluster Validation`: indicates the validity of the current `AMKOCluster` object.
+  * `member cluster initialisation`: indicates whether the cluster contexts given in `spec.clusters` were fetched and initialised from the `gslb-config-secret` secret. If a member cluster given in `spec.clusters` is not found in the `gslb-config-secret`, this step would fail.
+  * `member cluster validation`: The federator validates all the member clusters in the `spec.clusters` list and indicates a success/error. Validation includes some sanity checks, version mismatch checks, leader checks etc.
+  * `GSLBConfig federation`: The federator indicates whether it was able to federate the `GSLBConfig` object to all the clusters in `spec.clusters` successfully.
+  * `GDP Federation`: The federator indicates whether it was able to federate the `GDP`/`GlobalDeploymentPolicy` object to all the clusters in `spec.clusters` successfully.
+
+**Note** that if `helm` is used to deploy AMKO, this Custom Resource will be installed, and the users have to provide these values via `values.yaml`.
+
+
+**Additional Notes**:
+* The federation set can be a subset of the overall member cluster set used for GSLB.
+* AMKO has to be deployed on all clusters in the federation set.
+* `spec.clusterContext` must contain the current cluster's context.
+* `spec.version` is compared against the versions of all member clusters. If there's a version mismatch, the current AMKO will error out and the AMKO federator won't federate.
+* `spec.clusters` contains the federation cluster set.
+* Only one AMKO can be a leader, all other AMKOs have to be followers. If there are two leaders at any point, federation will stop and the error will be written to the `AMKOCluster`'s status.
+
+### Disasters and Recovery
+Whenever a disaster occurs and it takes down the cluster where the leader AMKO is deployed, the kubernetes/openshift objects will not be synced to the Avi GSLB Leader Controller anymore. At this point, the user has to switch one of the follower AMKOs to leader and the syncing should just resume.
```

Comment:
Consider changing to:
During a cluster down event on the `leader` AMKO, the federation of config objects will stop. However, at this point, all other clusters participating in federation would be synced with up to date configuration of the erstwhile GSLB `leader`. Hence, switching to a new AMKO `leader` does not require any manual steps of recovering the GSLB config objects.